pkg/remotedialer/server.go

```diff
@@ -63,7 +63,8 @@ func (s *Server) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	wsConn, err := upgrader.Upgrade(rw, req, nil)
 	if err != nil {
-		s.errorWriter(rw, req, 400, errors.Wrapf(err, "Error during upgrade for host [%v]", clientKey))
+		// Upgrade method takes care of returning the error to the client
+		logrus.WithError(err).Errorf("Error during upgrade for host [%v]", clientKey)
```

Comment:
I think that we still have to write response to the client. Can you log the error but also keep the original errorWriter in the server?

Reply:
Any error occurring during the connection upgrade has at this point already been written to the client by the `Upgrade()` method. See: https://github.com/rancher/norman/blob/5217a0efca14080d205d518ec9e25a14f4180d5c/vendor/github.com/gorilla/websocket/server.go#L51-L59
That's the reason why currently we see the following warning being logged by Go HTTP lib everytime the upgrade fails:
> http: multiple response.WriteHeader calls